analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ReturnValueIgnored.CSharp11.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+int x = 2;
+
+nint n = 2;
+nuint nu = 2;
+
+IntPtr intPtr = 2;
+UIntPtr uintPtr = 2;
+
+n.ToInt32(); // Compliant, FN, nint/IntPtr not considered an immutable type by the rule
```

Comment:
I would just mention that the function is `pure`, since this rule context can be also at the function level